controller/hydrator/hydrator.go

```diff
@@ -324,18 +360,18 @@ func (h *Hydrator) hydrate(logCtx *log.Entry, apps []*appv1.Application, project
 		app := app
 		eg.Go(func() error {
 			_, pathDetails, err = h.getManifests(ctx, app, targetRevision, projects[app.Spec.Project])
+			mu.Lock()
+			defer mu.Unlock()
 			if err != nil {
-				return fmt.Errorf("failed to get manifests for app %q: %w", app.QualifiedName(), err)
+				errors[app.QualifiedName()] = fmt.Errorf("failed to get manifests: %w", err)
+				return errors[app.QualifiedName()]
```

Comment:
If I'm reading line 374 correctly, this returned error goes unused. We always rely on the map instead.

If I'm right, then I think we should switch from an errgroup to a plain old wait group.

Reply:
Seems like a refactor outside the goal of this PR. There's also no unit test on the parallelism behavior, so a refactor could be in it's own PR.